spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderTests.java

```diff
@@ -479,10 +479,11 @@ public void testWellKnownHeaderMapperWithBeanNameKafkaHeaderMapper() throws Exce
 		}
 		KafkaTestBinder binder = new KafkaTestBinder(binderConfiguration, kafkaTopicProvisioner);
 		((GenericApplicationContext) binder.getApplicationContext()).registerBean("kafkaBinderHeaderMapper",
-				KafkaHeaderMapper.class, () -> new KafkaHeaderMapper() {
+				KafkaHeaderMapper.class, () -> new BinderHeaderMapper() {
 					@Override
 					public void fromHeaders(MessageHeaders headers, Headers target) {
 						target.add(new RecordHeader("custom-header", "foobar".getBytes()));
+						super.fromHeaders(headers, target);
```

Comment:
Well, how have we lived without this before?
What made you to add this code here?

I thought we need to do some fix which won't make us to change tests...

Reply:
:-), ya, we missed it before and these changes forced me to validate the fix and decided to make this change. Sorry.

Reply:
No problem! I just feel discomfort with a new code without any reasoning in the commit message...
Especially when I'm not an owner of the code base 😉